diff -x '*~' -rupN ../oww_client-0.5.9/src/owwl.c ../oww_client-0.5.9-jmc/src/owwl.c
--- ../oww_client-0.5.9/src/owwl.c	2012-04-11 17:47:53.000000000 -0400
+++ ../oww_client-0.5.9-jmc/src/owwl.c	2012-10-10 12:25:09.000000000 -0400
@@ -1207,6 +1207,7 @@ owwl_append_new_device(owwl_conn *conn, 
 
     case OwwlDev_Wind:
       conn->data[conn->data_count].str = string_from_wind ;
+      conn->data[conn->data_count].val = readout_wind ;
       break ;
 
     case OwwlDev_Rain:
@@ -1917,7 +1918,7 @@ owwl_decode(
       uvi->device_data.uvi.uvi = data_float[0] ;
 #      ifdef OWWL_DEBUG
         printf("Received UVI data: %f\n",
-          sol->device_data.uvi.uvi) ;
+          uvi->device_data.uvi.uvi) ;
 #      endif
       break ;
     }
@@ -2943,7 +2944,6 @@ void gettimeofday(struct timeval* pTv, v
 
   pTv->tv_sec  = ts.tv_sec;
   pTv->tv_usec = ts.tv_nsec / 1000;
-}
 #endif /* HAVE_GETTIMEOFDAY */
 #endif /* _WIN32 */
 
diff -x '*~' -rupN ../oww_client-0.5.9/src/owwl.h ../oww_client-0.5.9-jmc/src/owwl.h
--- ../oww_client-0.5.9/src/owwl.h	2012-04-11 17:21:19.000000000 -0400
+++ ../oww_client-0.5.9-jmc/src/owwl.h	2012-10-05 12:49:33.000000000 -0400
@@ -500,7 +500,7 @@ typedef union {
  * Functions to convert data to strings or doubles are provided.
  */
 typedef struct owwl_data_ {
-  int device_type ; /**< The type of device held in the struct */
+  enum owwl_device_type_enum device_type; //  int device_type ; /**< The type of device held in the struct */
   device_data_union device_data ;   /**< A union of all possible device data structures */
   int device_subtype     ; /**< Some devices have sub-types. For example, a
                                 temperature might come from a hygrometer */
